src/py/flwr/common/record/configsrecord.py

```diff
@@ -58,27 +58,57 @@ def is_valid(__v: ConfigsScalar) -> None:
 
 
 class ConfigsRecord(TypedDict[str, ConfigsRecordValues]):
-    """Configs record."""
+    """Configs record.
+
+    A :code:`ConfigsRecord` is a Python dictionary designed to ensure that
+    each key-value pair adheres to specified data types.
+
+    Parameters
+    ----------
+    configs_dict : Optional[Dict[str, ConfigsRecordValues]]
+        A dictionary that stores basic types (i.e. `str`, `int`, `float`, `bytes` as
+        defined in `ConfigsScalar`) and lists of such types (see
+        `ConfigsScalarList`).
+    keep_input : bool (default: True)
+        A boolean indicating whether config passed should be deleted from the input
+        dictionary immediately after adding them to the record. When set
+        to True, the data is duplicated in memory. If memory is a concern, set
+        it to False.
+
+    Examples
+    --------
+    The usage of a :code:`ConfigsRecord` is envisioned for storing Python built-in
+    types such as :code:`int`, :code:`float`, :code:`str`, :code:`bytes` and lists of
+    these. All types allowed are defined in
```

Comment:
It might be better to not only tell the user what they can technically do with `ConfigsRecord`, but what it is intended for. It's intended to send configuration values to the client, along with the model. Perhaps we can give a few examples, like sending the number of local epochs the client should perform? Or time-related things?